PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStorageCtrl.kt

```diff
@@ -148,22 +150,79 @@ abstract class SparkSubmissionJobUploadStorageCtrl(val view: SparkSubmissionJobU
     }
 
     private fun registerStorageInfoCheck(): Subscription = view.storageCheckSubject
-            .throttleWithTimeout(500, TimeUnit.MILLISECONDS)
-            .doOnNext { log().debug("Receive checking message $it") }
-            .flatMap { validateStorageInfo(it) }
+            .groupBy { checkEvent -> checkEvent::class.java.typeName}
             .subscribe(
-                    { },
+                    { groupedOb ->
+                        groupedOb
+                                .throttleWithTimeout(200, TimeUnit.MILLISECONDS)
+                                .doOnNext { log().info("Receive checking message ${it.message}") }
+                                .flatMap { validateStorageInfo(it) }
+                                .subscribe()
+                    },
                     { err -> log().warn(ExceptionUtils.getStackTrace(err)) })
 
     abstract fun getClusterDetail(): IClusterDetail?
 
-    open fun setDefaultStorageType(checkEvent:StorageCheckEvent) {
-        if (checkEvent is StorageCheckSelectedClusterEvent) {
-            val clusterDetail = getClusterDetail()
-            if (clusterDetail is HDInsightLivyLinkClusterDetail) {
-                view.storagePanel.storageTypeComboBox.selectedItem = view.storagePanel.sparkInteractiveSessionCard.title
-            } else {
-                view.storagePanel.storageTypeComboBox.selectedItem = view.storagePanel.clusterDefaultStorageCard.title
+    fun setDefaultStorageType(checkEvent:StorageCheckEvent) {
+        synchronized(view.storagePanel) {
+            if (checkEvent is StorageCheckSelectedClusterEvent) {
+
+                //check cluster type then reset storage combox
+                val defaultStorageTitle = view.storagePanel.clusterDefaultStorageCard.title
+                val sessionTitle = view.storagePanel.sparkInteractiveSessionCard.title
```

Comment:
`helpSessionTitle` may help remember it.

```suggestion
                val helpSessionTitle = view.storagePanel.sparkInteractiveSessionCard.title
```